flocker/acceptance/endtoend/test_dockerplugin.py

```diff
@@ -48,6 +48,25 @@ def get_path(name):
         return Client(base_url="https://{}:{}".format(address, DOCKER_PORT),
                       tls=tls, timeout=100)
 
+    def restart_docker(self, address):
+        """
+        Restart the Docker daemon on the specified address.
+
+        :param bytes address: The public IP of the node on which Docker will
+            be restarted.
+        """
+        try:
+            distro = run_ssh_command(
+                b"root", address, ["python", "-m", "platform"])
+            distro = distro.output.lower()
+            if 'ubuntu' in distro:
+                command = ["service", "docker", "restart"]
```

Comment:
Another approach would be to try one and then if it fails with "command not found" try the other. The most important thing is if _neither_ command works, we get a good error message and a failed test.